src/sys/wasi/mod.rs

```diff
@@ -186,12 +187,14 @@ impl Selector {
         };
         if let Some(index) = subscriptions.iter().position(predicate) {
             subscriptions.swap_remove(index);
-            /* NOTE: can't have read and write subscriptions.
             // We might have two subscriptions (read and write).
-            if let Some(index) = subscriptions.iter().skip(index).position(predicate) {
-                subscriptions.swap_remove(index);
+            loop {
```

Comment:
I think this loop should be moved to be outside of the `if` statement on 188, which is the same as the contents of the loop.

Reply:
done